src/debug/injected/recorder.ts

```diff
@@ -92,24 +92,69 @@ export class Recorder {
     });
   }
 
-  private _buildSelector(node: Node): string {
-    const element = node as Element;
-    for (const attribute of ['data-testid', 'aria-label', 'id', 'data-test-id', 'data-test']) {
+  private _buildSelector(targetElement: Element): string {
+    const path: string[] = [];
+    const root = document.documentElement;
+    for (let element: Element | null = targetElement; element && element !== root; element = element.parentElement) {
+      const selector = this._buildSelectorCandidate(element);
+      if (selector)
+        path.unshift(selector.selector);
+
+      const fullSelector = path.join(' ');
+      if (selector && selector.final)
+        return fullSelector;
+      if (targetElement === this._injectedScript.querySelector(parseSelector(fullSelector), root))
+        return fullSelector;
+    }
+    return '<selector>';
+  }
+
+  private _buildSelectorCandidate(element: Element): { final: boolean, selector: string } | null {
+    for (const attribute of ['data-testid', 'aria-label', 'data-test-id', 'data-test']) {
```

Comment:
Why aria-label is final? I thought it's similar to text.

Reply:
Dunno, just feels like a much stronger signal :/